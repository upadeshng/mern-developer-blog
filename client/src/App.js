import React from 'react';
import './App.css';

import { Provider } from 'react-redux';
import store from './redux/store';
import Routing from './routes';
// import { loadUser } from './component/auth/action';
// import setAuthToken from './utils/setAuthToken';

// if (localStorage.token) {
// setAuthToken(localStorage.token);
// }

const App = () => {
  return (
    <Provider store={store}>
      <Routing />
    </Provider>
  );
};

export default App;
