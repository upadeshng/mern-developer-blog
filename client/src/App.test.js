import React from 'react';
import App from './App';

import { shallow, mount, render } from 'enzyme';

import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

// https://www.robinwieruch.de/react-connected-component-test

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  //   it('should render with given state from Redux store', () => {
  //     expect(component.toJSON()).toMatchSnapshot();
  //   });
});

describe('App', () => {
  //   const wrapper = shallow(<Register />);
  let store = mockStore({
    myState: 'sample text',
  });

  const wrapper = shallow(
    <Provider store={store}>
      // Provides the store to your App
      <App />
    </Provider>
  );

  //   Snapshot
  it('Renders correctly with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
