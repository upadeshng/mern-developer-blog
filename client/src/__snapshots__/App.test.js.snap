// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders correctly with snapshot 1`] = `ShallowWrapper {}`;
